File: src/main/java/org/redrune/net/packet/incoming/impl/CapeColorCustomizationPacketReader.java
package org.redrune.net.packet.incoming.impl;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.net.packet.Packet;
import org.redrune.net.packet.context.PacketContext;
import org.redrune.net.packet.context.impl.CapeColorCustomizationPacketContext;
import org.redrune.net.packet.incoming.IncomingPacketReader;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
public class CapeColorCustomizationPacketReader implements IncomingPacketReader {
	
	@Override
	public int[] bindings() {
		return arguments(COLOR_ID_PACKET);
	}
	
	@Override
	public PacketContext read(Player player, Packet packet) {
		int colorId = packet.readUnsignedShort();
		return new CapeColorCustomizationPacketContext(colorId);
	}
}

File: src/main/java/org/redrune/networking/packet/incoming/impl/CapeColorCustomizationPacketReader.kt
package org.redrune.networking.packet.incoming.impl

import org.redrune.game.entity.actor.player.Player
import org.redrune.networking.packet.Packet
import org.redrune.networking.packet.context.PacketContext
import org.redrune.networking.packet.context.impl.CapeColorCustomizationPacketContext
import org.redrune.networking.packet.incoming.IncomingPacketReader
import org.redrune.utility.constants.PacketConstants

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
class CapeColorCustomizationPacketReader : IncomingPacketReader {
    override fun bindings(): IntArray {
        return arguments(PacketConstants.COLOR_ID_PACKET)
    }

    override fun read(player: Player, packet: Packet): PacketContext {
        val colorId = packet.readUnsignedShort()
        return CapeColorCustomizationPacketContext(colorId)
    }
}